core/types/transaction.go

```diff
@@ -379,13 +379,7 @@ func (tx *Transaction) RollupDataGas() RollupGasData {
 		log.Error("failed to encode tx for L1 cost computation", "err", err)
 	}
 	var out RollupGasData
```

Comment:
What about a `NewRollupGasData(data)` pattern? Since the setter is always required, this pattern reduces the chances of forgetting